gma/integration_test/src/integration_test.cc

```diff
@@ -845,16 +842,30 @@ TEST_F(FirebaseGmaTest, TestRewardedAdLoad) {
   firebase::Future<firebase::gma::AdResult> load_ad_future =
       rewarded->LoadAd(kRewardedAdUnit, GetAdRequest());
 
-  WaitForCompletion(load_ad_future, "LoadAd");
-  const firebase::gma::AdResult* result_ptr = load_ad_future.result();
-  ASSERT_NE(result_ptr, nullptr);
-  EXPECT_TRUE(result_ptr->is_successful());
-  EXPECT_FALSE(result_ptr->response_info().adapter_responses().empty());
-  EXPECT_FALSE(
-      result_ptr->response_info().mediation_adapter_class_name().empty());
-  EXPECT_FALSE(result_ptr->response_info().response_id().empty());
-  EXPECT_FALSE(result_ptr->response_info().ToString().empty());
-
+  // This test behaves differently if it's running in UI mode
+  // (manually on a device) or in non-UI mode (via automated tests).
+  if (ShouldRunUITests()) {
+    // Run in manual mode: fail if any error occurs.
+    WaitForCompletion(load_ad_future, "LoadAd");
+  } else {
+    // Run in automated test mode: don't fail if NoFill occurred.
+    WaitForCompletionAnyResult(load_ad_future,
+                               "LoadAd (ignoring NoFill error)");
+    EXPECT_TRUE(load_ad_future.error() == firebase::gma::kAdErrorCodeNone ||
+                load_ad_future.error() == firebase::gma::kAdErrorCodeNoFill);
+  }
+  if (load_ad_future.error() == firebase::gma::kAdErrorCodeNone) {
```

Comment:
The problem with this solution is that the test will generate NoFill errors eventually. On future runs the other non-stress tests will start to get either NoFill errors or worse: Backoff errors.

I think it's better to simply disable these in CI and have them as part of our manual testing.

Reply:
I take it back, I thought this was in the stress test. IGNORE ME!